lib/synapse/service_watcher/multi/resolver/base.rb

```diff
@@ -0,0 +1,38 @@
+require "synapse/log"
+require "synapse/statsd"
+
+class Synapse::ServiceWatcher::Resolver
+  class BaseResolver
+    include Synapse::Logging
+    include Synapse::StatsD
+
+    def initialize(opts, watchers)
+      super()
+
+      log.info "creating base resolver"
+
+      raise ArgumentError, "BaseResolver expects method to be base" unless opts['method'] == 'base'
+      raise ArgumentError, "no watchers provided" unless watchers.length > 0
+    end
+
+    # should be overridden in child classes
+    def start
+      log.info "starting base resolver"
```

Comment:
throw exception here if you want to force child class to implement it. Similar for other methods.

Reply:
The current `BaseWatcher` does not throw an exception; instead, the `BaseWatcher` is actually usable as-is; that allows you to run Synapse without any complicated setup (i.e. no Zookeeper or anything like that) by using the base watcher/resolver. I wanted to preserve that property